src/rpdk/core/contract/resource_client.py

```diff
@@ -133,6 +133,15 @@ def has_writable_identifier(self):
                     return True
         return False
 
+    def assert_write_only_property_does_not_exist(self, resource_model):
+        if self.write_only_paths:
+            assert not any(
+                traverse(
```

Comment:
if this attempts to fetch a key that does exist does it continue or throw an exception?

Reply:
This test was not set up right. I have fixed it now to pass None value to write_only property

Reply:
to clarify if a writeOnlyProperty doesn't exist what do we return? it looks like traverse will throw a KeyError:

https://github.com/aws-cloudformation/cloudformation-cli/blob/71e17edee7193fc4bbf17940ed4d4905ca81df8a/src/rpdk/core/jsonutils/utils.py#L82-L85

Reply:
I am checking ```if self.write_only_paths:``` at line 137. We should not get the Key Error.